apps/registry/app/api/chat/route.ts

```
@@ -9,11 +9,21 @@ export const runtime = "edge";
+  const { messages, system, tools } = await req.json();
 
   const result = streamText({
     model: openai("gpt-4o"),
     messages,
+    // forward system prompt and tools from the frontend
+    system,
+    tools: Object.fromEntries(
+      Object.entries<{ parameters: unknown }>(tools).map(([name, tool]) => [
+        name,
+        {
+          parameters: jsonSchema(tool.parameters!),
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)